<a href="https://colab.research.google.com/github/ranivaradharajcp31-png/youtube_monetization_model_ml/blob/main/content_monetization_modeler.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd
data_path="youtube_ad_revenue_dataset.csv"
df=pd.read_csv(data_path)
df.head()

,video_id,date,views,likes,comments,watch_time_minutes,video_length_minutes,subscribers,category,device,country,ad_revenue_usd
0,vid_3092,2024-09-24 10:50:40.993199,9936,1221.0,320.0,26497.214184,2.862137,228086,Entertainment,TV,IN,203.178237
1,vid_3459,2024-09-22 10:50:40.993199,10017,642.0,346.0,15209.747445,23.738069,736015,Gaming,Tablet,CA,140.880508
2,vid_4784,2024-11-21 10:50:40.993199,10097,1979.0,187.0,57332.658498,26.200634,240534,Education,TV,CA,360.134008
3,vid_4078,2025-01-28 10:50:40.993199,10034,1191.0,242.0,31334.517771,11.770340,434482,Entertainment,Mobile,UK,224.638261
4,vid_3522,2025-04-28 10:50:40.993199,9889,1858.0,477.0,15665.666434,6.635854,42030,Education,Mobile,CA,165.514388


In [6]:
df.shape
print("Rows and columns = ", df.shape)

Rows and columns =  (122400, 12)


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 122400 entries, 0 to 122399
Data columns (total 12 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   video_id              122400 non-null  object 
 1   date                  122400 non-null  object 
 2   views                 122400 non-null  int64  
 3   likes                 116283 non-null  float64
 4   comments              116288 non-null  float64
 5   watch_time_minutes    116295 non-null  float64
 6   video_length_minutes  122400 non-null  float64
 7   subscribers           122400 non-null  int64  
 8   category              122400 non-null  object 
 9   device                122400 non-null  object 
 10  country               122400 non-null  object 
 11  ad_revenue_usd        122400 non-null  float64
dtypes: float64(5), int64(2), object(5)
memory usage: 11.2+ MB


In [7]:
df.describe()

,views,likes,comments,watch_time_minutes,video_length_minutes,subscribers,ad_revenue_usd
count,122400.000000,116283.000000,116288.000000,116295.000000,122400.000000,122400.000000,122400.000000
mean,9999.856283,1099.633618,274.396636,37543.827721,16.014165,502191.719902,252.727210
std,99.881260,519.424089,129.741739,12987.724246,8.083790,288397.470103,61.957052
min,9521.000000,195.000000,48.000000,14659.105562,2.000142,1005.000000,126.590603
25%,9933.000000,650.000000,162.000000,26366.320569,9.004695,252507.500000,199.902018
50%,10000.000000,1103.000000,274.000000,37531.990337,16.005906,503465.500000,252.749699
75%,10067.000000,1547.000000,387.000000,48777.782090,23.021260,752192.000000,305.597518
max,10468.000000,2061.000000,515.000000,61557.670089,29.999799,999997.000000,382.768254


## Initial observations

- 122,400 rows, 12 columns. Target: `ad_revenue_usd`.
- Three columns have missing values: `likes`, `comments`, `watch_time_minutes` (~5% each).
- No negative or zero values anywhere — no rows to remove.
- `views` has a standard deviation of only 1% of its mean (9,521 to 10,468).
  It is nearly constant, so it is unlikely to be a useful predictor.
- Mean equals median in every numeric column — perfectly symmetric, no skew.
  Real revenue data is right-skewed, which confirms this dataset is synthetic.

In [8]:
print("Missing values per column:")
print(df.isnull().sum())

print("\nAs a percentage of all rows:")
print((df.isnull().sum() / len(df) * 100).round(2))

print("\nExact duplicate rows:", df.duplicated().sum())

Missing values per column:
video_id                   0
date                       0
views                      0
likes                   6117
comments                6112
watch_time_minutes      6105
video_length_minutes       0
subscribers                0
category                   0
device                     0
country                    0
ad_revenue_usd             0
dtype: int64

As a percentage of all rows:
video_id                0.00
date                    0.00
views                   0.00
likes                   5.00
comments                4.99
watch_time_minutes      4.99
video_length_minutes    0.00
subscribers             0.00
category                0.00
device                  0.00
country                 0.00
ad_revenue_usd          0.00
dtype: float64

Exact duplicate rows: 2400


In [9]:
before = df.shape[0]
df = df.drop_duplicates()
after = df.shape[0]

print(f"Rows before : {before}")
print(f"Rows after  : {after}")
print(f"Removed     : {before - after} duplicate rows")

Rows before : 122400
Rows after  : 120000
Removed     : 2400 duplicate rows
